cmd/fetch_commit.go

```diff
@@ -79,13 +80,18 @@ Use public key from a kubernetes secret:
 			return nil
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
+			invalidPredicateMaterialCount := 0
 			for _, comp := range data.spec.Components {
 				err := validateGitSource(cmd.Context(), comp.ContainerImage, data.publicKey)
 				if err != nil {
 					log.Println(err)
+					invalidPredicateMaterialCount += 1
```

Comment:
This is counting if we got a git source for each component, which seem like a reasonable thing to do. However, this is not the problem outlined in [HACBS-1089](https://issues.redhat.com/browse/HACBS-1089). 

There could be, and there will always be in our use cases, multiple attestations per component/image. So for each image, we want to make sure that at least one of the attestations contains the git source information.

The check, I believe, should be added somewhere [here](https://github.com/hacbs-contract/ec-cli/blob/1162bdbd0e01426f84275faed47a0a11342e2cbb/cmd/fetch_commit.go#L107-L135).

Here's my reproducer:

```
🐚 cat cosign.pub 
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEe/UANElkGi2vnQsVr1rGOKfzmxkR
34RSOTNutukkeDmhFsHvI7BYEaii9wlWvsjDqc1Nt2IHxp9CKHQafRH4Vw==
-----END PUBLIC KEY-----

🐚 go run main.go fetch commit --image-ref quay.io/lucarval/tekton-test:latest --public-key cosign.pub
fetch_commit.go:87: there is no authorization source in attestation. sha: , url: 
Error: no valid .predicate.materials found in attestations
exit status 1
```

The image has two attestations, only one of which contains the git source information:
```
🐚 cosign verify-attestation --key cosign.pub quay.io/lucarval/tekton-test:latest 2> /dev/null | jq '.payload | @base64d | fromjson | .predicate.materials'
null
[
  {
    "uri": "git+https://github.com/lcarva/minimal-container.git",
    "digest": {
      "sha1": "10ab45329f9f29a2c84379b86f5b7e98a54f250e"
    }
  }
]
```